src/Param_TEST.cc

```diff
@@ -75,11 +77,6 @@ TEST(Param, Bool)
   strParam.Get<bool>(value);
   EXPECT_TRUE(value);
 
-  // Anything other than 1 or true is treated as a false value
```

Comment:
Is there a reason for removing this test?

Reply:
Since this section was removed:
https://github.com/ignitionrobotics/sdformat/blob/f3dbe1d225939ff1bc038b0789fff884d763cfa7/include/sdf/Param.hh#L296-L315

It now reaches this error: https://github.com/ignitionrobotics/sdformat/blob/f0f9759ec9d4572db9c5fd9e0dd861e6f679a7cf/src/Param.cc#L329

To me, it made more sense to error when a random value such as `%` was set rather than defaulting to `false`. Should I change this back?

Reply:
How's this? 650b544c92c2e1abfb4b7adf6198e7a21b97187f

This seemed to be the easiest fix. Changing `ParamPrivate::ValueFromStringImpl` bool section to assign the value regardless of the error and return true (since `_valueToSet` was successfully set) caused `Param.InvalidBool` test to fail

Reply:
That works. Can you add a comment there saying that the piece of code handling bool types is there to keep backward behavior?

I'm thinking we should change the behavior for Fortress. Would you agree?

Reply:
I agree. I'll add the comment and put a `TODO` to remove for Fortress

Reply:
d3532946bc64fef713b30fbb1ab21a33603bf9dd